action.yml

```diff
@@ -75,59 +67,197 @@ runs:
+            MAJOR_BRANCH=$(echo "$MAJOR_BRANCHES" | head -2 | tail -1)
           fi
-        elif [[ $major_type == "previous" ]]; then
-          if (( $major_branches_count < 2 )); then
+        elif [[ $MAJOR_TYPE == "previous" ]]; then
+          if (( $MAJOR_BRANCHES_COUNT < 2 )); then
             echo "Only one major branch found, not running CI on this cycle - will run on next cycle"
             exit 0
           fi
-          major_branch=$(echo "$major_branches" | head -2 | tail -1)
-          if (( $latest_major_stable_tags_count == 0 )) && (( $major_branches_count >= 3 )); then
-            major_branch=$(echo "$major_branches" | head -3 | tail -1)
+          MAJOR_BRANCH=$(echo "$MAJOR_BRANCHES" | head -2 | tail -1)
+          if (( $LATEST_MAJOR_STABLE_TAGS_count == 0 )) && (( $MAJOR_BRANCHES_COUNT >= 3 )); then
+            MAJOR_BRANCH=$(echo "$MAJOR_BRANCHES" | head -3 | tail -1)
           fi
         fi
-        if [[ $minor_type == "next-patch" ]]; then
-          minor_branches=$(jq -r ".[] | select(.name | test(\"^$major_branch.[0-9]+$\")) | .name" __branches.json | sort -V -r)
-          minor_branches_count=$(echo "$minor_branches" | wc -l)
-          if (( $minor_branches_count == 0 )); then
-            echo "No minor branches found on major branch $major_branch when minor_type is next-patch, cannot continue"
+        if [[ $MINOR_TYPE == "next-patch" ]]; then
+          MINOR_BRANCHES=$(jq -r --arg major_branch "$MAJOR_BRANCH" ".[] | select(.name | test('^$major_branch.[0-9]+$')) | .name" __branches.json | sort -V -r)
+          MINOR_BRANCHES_COUNT=$(echo "$MINOR_BRANCHES" | wc -l)
+          if (( $MINOR_BRANCHES_COUNT == 0 )); then
+            echo "No minor branches found on major branch $MAJOR_BRANCH when MINOR_TYPE is next-patch, cannot continue"
             exit 1
           fi
-          minor_branch=$(echo "$minor_branches" | head -1)
+          MINOR_BRANCH=$(echo "$MINOR_BRANCHES" | head -1)
         fi
-        branch=$major_branch
-        if [[ $minor_branch != "" ]]; then
-          branch="$minor_branch"
+        echo "MAJOR_BRANCH is $MAJOR_BRANCH"
+        echo "major_branch=$MAJOR_BRANCH" >> $GITHUB_OUTPUT
+        BRANCH=$MAJOR_BRANCH
+        if [[ $MINOR_BRANCH != "" ]]; then
+          BRANCH="$MINOR_BRANCH"
         fi
-        echo "branch is $branch"
-        echo "branch=$branch" >> $GITHUB_OUTPUT
+        echo "BRANCH is $BRANCH"
+        echo "BRANCH=$BRANCH" >> $GITHUB_OUTPUT
+
+    - name: Derive should dispatch
+      id: shoulddispatch
+      shell: bash
+      env:
+        MODULE_MAP: ${{ steps.modulemap.outputs.module_map }}
+        FRAMEWORK_TWO_TAG_EXISTS: ${{ steps.frameworkstatus.outputs.framework_two_tag_exists }}
+        MAJOR_BRANCH: ${{ steps.getbranch.outputs.major_branch }}
+        MAJOR_TYPE: ${{ steps.gettype.outputs.major_type }}
+        BRANCH: ${{ steps.branches.outputs.branch }}
+      run: |
+        SHOULD_DISPATCH=1
+        if [[ $BRANCH == "" ]]; then
```

Comment:
This is to handle legit `exit 0` style missing branches that we want to go green.  e.g. currently only a single major branch

There are some `exit 1` go red style conditions in the block above e.g. no minor branches found on previous major branch